web-api/src/persistence/postgres/cases/getConsolidatedCases.ts

```diff
@@ -0,0 +1,59 @@
+import { Case } from '@shared/business/entities/cases/Case';
+import { getDbReader } from '@web-api/database';
+import {
+  ALL_OMITTABLE_CASE_FIELDS,
+  getCasesByDocketNumbers,
+  OmittableCaseFields,
+} from '@web-api/persistence/postgres/cases/getCasesByDocketNumbers';
+import { fromKyselyCase } from '@web-api/persistence/postgres/cases/mapper';
+
+export const getConsolidatedCases = async <
+  T extends OmittableCaseFields[] = [],
+>({
+  leadDocketNumber,
+  excludeFields,
+}: {
+  leadDocketNumber: string;
+  excludeFields?: T;
+}): Promise<Omit<RawCase, 'consolidatedCases' | T[number]>[]> => {
+  // If we only need case metadata, make one query for all of it
```

Comment:
The tradeoff:

1. Slightly simpler code (always make two db requests, one to get docket numbers and one to `getCasesByDocketNumbers`) that is slightly less efficient sometimes
2. Slightly more complex code (only make one db query in cases we can do so) that is slightly more efficient sometimes

Implemented here is option 2.